### DISCUSSION TITLE
如何实现各种类型(如 int,float,stringRef )与Value*之间的类型转换？

### DISCUSSION THREAD

**Discussion by Aurora-1412** [arcsysu/SYsU-lang#39] [Q&A] [answered]
看API看得有点晕emmm，没找到

**Comment by Yorkking** [comment, marked as answer]
RTM: https://llvm.org/doxygen/Type_8h_source.html

**Reply by Aurora-1412** [reply]
> RTM: https://llvm.org/doxygen/Type_8h_source.html

这里只有int等类型到Type*的转换emmm

**Reply by Yorkking** [reply]
实验中的样例基本都是 int 哇，不用考虑那么复杂
![image](https://user-images.githubusercontent.com/34130042/168831884-8351d54d-97df-4cd5-b062-9e447ca408dd.png)

**Reply by Aurora-1412** [reply]
> 

我在纠结怎么把int类型转成Value*类型QAQ

**Reply by Yorkking** [reply]
```C
llvm::ConstantInt::get(
          TheContext, /* i32 3(decimal) */ llvm::APInt(32, string_val, 10)); // string_val 为 str 这样就行了。
```

**Comment by Aurora-1412** [comment]
谢谢|･ω･｀) 刚才傻了，constantInt*和Value*应该是可以自动隐式类型转换的



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2022年5月17日(周二) 晚上10:17
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [arcsysu/SYsU-lang] 如何实现各种类型(如 int,float,stringRef )与Value*之间的类型转换？ (Discussion #39)




 llvm::ConstantInt::get(           TheContext, /* i32 3(decimal) */ llvm::APInt(32, string_val, 10)); // string_val 为 str 这样就行了。
